include/oneapi/dpl/experimental/kt/single_pass_scan.h

```diff
@@ -0,0 +1,434 @@
+          typename... _Name>
+struct __lookback_submitter<__data_per_workitem, __workgroup_size, _Type, _FlagType,
+                            oneapi::dpl::__par_backend_hetero::__internal::__optional_kernel_name<_Name...>>
+{
+
+    template <typename _InRng, typename _OutRng, typename _BinaryOp, typename _StatusFlags, typename _StatusValues>
+    sycl::event
+    operator()(sycl::queue __q, sycl::event __prev_event, _InRng&& __in_rng, _OutRng&& __out_rng, _BinaryOp __binary_op,
+               ::std::size_t __n, _StatusFlags&& __status_flags, ::std::size_t __status_flags_size,
+               _StatusValues&& __status_vals_full, _StatusValues&& __status_vals_partial,
+               ::std::size_t __current_num_items) const
+    {
+        using _LocalAccessorType = sycl::local_accessor<_Type, 1>;
+        using _KernelFunc =
+            __lookback_kernel_func<__data_per_workitem, __workgroup_size, _Type, _FlagType, ::std::decay_t<_InRng>,
+                                   ::std::decay_t<_OutRng>, ::std::decay_t<_BinaryOp>, ::std::decay_t<_StatusFlags>,
+                                   ::std::decay_t<_StatusValues>, ::std::decay_t<_LocalAccessorType>>;
+        using _KernelName = __lookback_kernel<_Name..., _KernelFunc>;
+
+        static constexpr std::uint32_t __elems_in_tile = __workgroup_size * __data_per_workitem;
+
+        return __q.submit([&](sycl::handler& __hdl) {
+            auto __tile_vals = _LocalAccessorType(sycl::range<1>{__elems_in_tile}, __hdl);
+            __hdl.depends_on(__prev_event);
+
+            oneapi::dpl::__ranges::__require_access(__hdl, __in_rng, __out_rng);
+            __hdl.parallel_for<_KernelName>(sycl::nd_range<1>(__current_num_items, __workgroup_size),
+                                            _KernelFunc{__in_rng, __out_rng, __binary_op, __n, __status_flags,
+                                                        __status_flags_size, __status_vals_full, __status_vals_partial,
+                                                        __current_num_items, __tile_vals});
+        });
+    }
+};
+
+template <bool _Inclusive, typename _InRange, typename _OutRange, typename _BinaryOp, typename _KernelParam>
+sycl::event
+__single_pass_scan(sycl::queue __queue, _InRange&& __in_rng, _OutRange&& __out_rng, _BinaryOp __binary_op, _KernelParam)
+{
+    using _Type = oneapi::dpl::__internal::__value_t<_InRange>;
+    using _FlagType = __scan_status_flag<_Type>;
+    using _FlagStorageType = typename _FlagType::_FlagStorageType;
+
+    using _KernelName = typename _KernelParam::kernel_name;
+    using _LookbackInitKernel =
+        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<__lookback_init_kernel<_KernelName>>;
+    using _LookbackKernel =
+        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<__lookback_kernel<_KernelName>>;
+
+    const ::std::size_t __n = __in_rng.size();
```

Comment:
Stylistic suggestion: it might be worth it to go ahead and convert `::std::` to `std::` in this PR's additions since it seems that will be our convention going forward.

Reply:
Sounds good. I went ahead and made that change in this file and the test.